### PR TITLE
#358 Fix not saving 'show_mode'

### PR DESCRIPTION
**Pull request by IvanKirpichnikov**
Fixed:
- #358

Added:
- New 'show_mode' attribute for 'Context'.
- New 'show_mode' argument for 'Stack.push' of type 'Optional[ShowMode]'

### CODE REVIEW

File: src/aiogram_dialog/api/entities/stack.py [hunk 1]
```diff
@@ -2,12 +2,15 @@
...
 import time
 from dataclasses import dataclass, field
-from typing import List, Optional
+from typing import List, Optional, TYPE_CHECKING
 
 from aiogram.fsm.state import State
 
 from aiogram_dialog.api.exceptions import DialogStackOverflow
 from .context import Context, Data
+if TYPE_CHECKING:
```
Comment (Tishka17) [inline src/aiogram_dialog/api/entities/stack.py:11]: please, no

File: src/aiogram_dialog/api/entities/context.py [hunk 2]
```diff
@@ -1,8 +1,12 @@
 from dataclasses import dataclass, field
-from typing import Dict, List, Union
+from typing import Dict, List, Optional, TYPE_CHECKING, Union
 
 from aiogram.fsm.state import State
 
+if TYPE_CHECKING:
+    from ...api.entities import ShowMode
```
Comment (Tishka17) [inline src/aiogram_dialog/api/entities/context.py:7]: look in other modules how do I import

File: src/aiogram_dialog/api/entities/stack.py (hunk 1 above)
Comment (IvanKirpichnikov) [reply src/aiogram_dialog/api/entities/stack.py]: Okay, fixed it

File: src/aiogram_dialog/api/entities/context.py (hunk 2 above)
Comment (IvanKirpichnikov) [reply src/aiogram_dialog/api/entities/context.py]: done

File: src/aiogram_dialog/api/entities/context.py [hunk 3]
```diff
@@ -15,6 +18,7 @@ class Context:
     start_data: Data = field(compare=False)
     dialog_data: DataDict = field(compare=False, default_factory=dict)
     widget_data: DataDict = field(compare=False, default_factory=dict)
+    show_mode: Optional[ShowMode] = field(default=None, compare=False)
```
Comment (Tishka17) [inline src/aiogram_dialog/api/entities/context.py:21]: do we really need to store show_mode between events?

**Pull request closed without merging**